File: Search/ECMSearchWPF/CS/app_code/clsCommonFunctions.cs
// VBConversions Note: VB project level imports
using System.Windows.Input;
using System.Windows.Data;
using System.Windows.Documents;
using System.Xml.Linq;
using System.Collections.Generic;
using System.Windows.Navigation;
using System.Windows.Media.Imaging;
using System.Windows;
using Microsoft.VisualBasic;
using System.Windows.Media;
using System.Collections;
using System;
using System.Windows.Shapes;
using System.Windows.Controls;
using System.Threading.Tasks;
using System.Linq;
using System.Diagnostics;
// End of VB project level imports


namespace ECMSearchWPF
{
	public class clsCommonFunctions
	{
		
		
		//Dim proxy As New SVCSearch.Service1Client
		clsLogging LOG = new clsLogging();
		string SecureID;
		//Dim EP As New clsEndPoint
		
		public void RemoveSingleQuotes(ref string tVal)
		{
			tVal = tVal.Replace("\'\'", "\'");
			tVal = tVal.Replace("\'", "\'\'");
		}
		
		public clsCommonFunctions()
		{
			SecureID = GLOBALS._SecureID.ToString();
		}
		
		public void SaveClick(int ID, string UID)
		{
			bool RC = true;
			GLOBALS.ProxySearch.SaveClickStatsCompleted += new System.EventHandler(client_SaveClickStats);
			//EP.setSearchSvcEndPoint(proxy)
			GLOBALS.ProxySearch.SaveClickStatsAsync(GLOBALS._SecureID, ID, UID, RC);
		}
		
		public void client_SaveClickStats(object sender, SVCSearch.SaveClickStatsCompletedEventArgs e)
		{
			if (e.Error == null)
			{
				bool RC = System.Convert.ToBoolean(e.RC);
				if (! RC)
				{
					LOG.WriteToSqlLog("ERROR client_SaveClickStats: Failed to update.");
				}
			}
			else
			{
				LOG.WriteToSqlLog((string) ("ERROR 100 client_SaveClickStats: " + e.Error.Message));
			}
			GLOBALS.ProxySearch.SaveClickStatsCompleted -= new System.EventHandler(client_SaveClickStats);
		}
		public void RecordGrowth()
		{
			bool RC = false;
			
			GLOBALS.ProxySearch.RecordGrowthCompleted += new System.EventHandler(client_RecordGrowth);
			//EP.setSearchSvcEndPoint(proxy)
			GLOBALS.ProxySearch.RecordGrowthAsync(SecureID, RC);
			
		}
		public void client_RecordGrowth(object sender, SVCSearch.RecordGrowthCompletedEventArgs e)
		{
			if (e.Error == null)
			{
				bool RC = System.Convert.ToBoolean(e.RC);
				if (! RC)
				{
					LOG.WriteToSqlLog("ERROR client_RecordGrowth: Failed to update the associated DB Growthss.");
				}
			}
			else
			{
				LOG.WriteToSqlLog("ERROR 100 client_RecordGrowth: Failed to update the associated DB Growth.");
			}
			GLOBALS.ProxySearch.RecordGrowthCompleted -= new System.EventHandler(client_RecordGrowth);
		}
		
		public void resetMissingEmailIds()
		{
			bool RC = false;
			
			GLOBALS.ProxySearch.resetMissingEmailIdsCompleted += new System.EventHandler(client_resetMissingEmailIds);
			//EP.setSearchSvcEndPoint(proxy)
			GLOBALS.ProxySearch.resetMissingEmailIdsAsync(SecureID, modGlobals.gCurrUserGuidID, RC);
			
		}
		public void client_resetMissingEmailIds(object sender, SVCSearch.resetMissingEmailIdsCompletedEventArgs e)
		{
			if (e.Error == null)
			{
				bool RC = System.Convert.ToBoolean(e.RC);
				if (! RC)
				{
					LOG.WriteToSqlLog("ERROR client_resetMissingEmailIds: Failed to update the associated record(s).");
				}
			}
			else
			{
				LOG.WriteToSqlLog("ERROR 100 client_resetMissingEmailIds: Failed to update the associated  record(s).");
			}
			GLOBALS.ProxySearch.resetMissingEmailIdsCompleted -= new System.EventHandler(client_resetMissingEmailIds);
		}
		
	}
	
}

File: Search/ECMSearchWPF/app_code/clsCommonFunctions.vb
' ***********************************************************************
' Assembly         : ECMSearchWPF
' Author           : wdale
' Created          : 06-28-2020
'
' Last Modified By : wdale
' Last Modified On : 06-28-2020
' ***********************************************************************
' <copyright file="clsCommonFunctions.vb" company="D. Miller and Associates, Limited">
'     Copyright @ DMA Ltd 2020 all rights reserved.
' </copyright>
' <summary></summary>
' ***********************************************************************
''' <summary>
''' Class clsCommonFunctions.
''' </summary>
Public Class clsCommonFunctions


    'Dim proxy As New SVCSearch.Service1Client
    ''' <summary>
    ''' The log
    ''' </summary>
    Dim LOG As New clsLogMain
    ''' <summary>
    ''' The g secure identifier
    ''' </summary>
    Dim gSecureID As String
    'Dim EP As New clsEndPoint

    ''' <summary>
    ''' Removes the single quotes.
    ''' </summary>
    ''' <param name="tVal">The t value.</param>
    Sub RemoveSingleQuotes(ByRef tVal As String)
        tVal = tVal.Replace("''", "'")
        tVal = tVal.Replace("'", "''")
    End Sub

    ''' <summary>
    ''' Initializes a new instance of the <see cref="clsCommonFunctions"/> class.
    ''' </summary>
    Sub New()
        gSecureID = _SecureID
    End Sub

    ''' <summary>
    ''' Saves the click.
    ''' </summary>
    ''' <param name="ID">The identifier.</param>
    ''' <param name="UID">The uid.</param>
    Public Sub SaveClick(ByVal ID As Integer, ByVal UID As String)
        Dim RC As Boolean = True
        'AddHandler ProxySearch.SaveClickStatsCompleted, AddressOf client_SaveClickStats
        'EP.setSearchSvcEndPoint(proxy)
        ProxySearch.SaveClickStats(_SecureID, ID, UID, RC)
        client_SaveClickStats(RC)
    End Sub

    ''' <summary>
    ''' Clients the save click stats.
    ''' </summary>
    ''' <param name="RC">if set to <c>true</c> [rc].</param>
    Sub client_SaveClickStats(RC As Boolean)
        If RC Then
            If Not RC Then
                LOG.WriteToSqlLog("ERROR client_SaveClickStats: Failed to update.")
            End If
        Else
            LOG.WriteToSqlLog("ERROR 100 client_SaveClickStats: ")
        End If
        ''RemoveHandler ProxySearch.SaveClickStatsCompleted, AddressOf client_SaveClickStats
    End Sub
    ''' <summary>
    ''' Records the growth.
    ''' </summary>
    Public Sub RecordGrowth()
        Dim RC As Boolean = False

        'AddHandler ProxySearch.RecordGrowthCompleted, AddressOf client_RecordGrowth
        'EP.setSearchSvcEndPoint(proxy)
        ProxySearch.RecordGrowth(gSecureID, RC)
        client_RecordGrowth(RC)
    End Sub
    ''' <summary>
    ''' Clients the record growth.
    ''' </summary>
    ''' <param name="RC">if set to <c>true</c> [rc].</param>
    Sub client_RecordGrowth(RC As Boolean)
        If RC Then
            If Not RC Then
                LOG.WriteToSqlLog("ERROR client_RecordGrowth: Failed to update the associated DB Growthss.")
            End If
        Else
            LOG.WriteToSqlLog("ERROR 100 client_RecordGrowth: Failed to update the associated DB Growth.")
        End If
        ''RemoveHandler ProxySearch.RecordGrowthCompleted, AddressOf client_RecordGrowth
    End Sub

    ''' <summary>
    ''' Resets the missing email ids.
    ''' </summary>
    Public Sub resetMissingEmailIds()
        Dim RC As Boolean = False

        'AddHandler ProxySearch.resetMissingEmailIdsCompleted, AddressOf client_resetMissingEmailIds
        'EP.setSearchSvcEndPoint(proxy)
        ProxySearch.resetMissingEmailIds(gSecureID, gCurrUserGuidID, RC)
        client_resetMissingEmailIds(RC)
    End Sub
    ''' <summary>
    ''' Clients the reset missing email ids.
    ''' </summary>
    ''' <param name="RC">if set to <c>true</c> [rc].</param>
    Sub client_resetMissingEmailIds(RC As Boolean)
        If RC Then
            If Not RC Then
                LOG.WriteToSqlLog("ERROR client_resetMissingEmailIds: Failed to update the associated record(s).")
            End If
        Else
            LOG.WriteToSqlLog("ERROR 100 client_resetMissingEmailIds: Failed to update the associated  record(s).")
        End If
        ''RemoveHandler ProxySearch.resetMissingEmailIdsCompleted, AddressOf client_resetMissingEmailIds
    End Sub

End Class
